sda/websocket.go

```diff
@@ -0,0 +1,97 @@
+package sda
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"time"
+
+	"github.com/dedis/cothority/log"
+	"github.com/dedis/cothority/network"
+	"golang.org/x/net/websocket"
+	"gopkg.in/tylerb/graceful.v1"
+)
+
+const readSize = 1024
+
+// WebSocket handles incoming client-requests using the WebService
+// protocol for JavaScript-compatibility.
+type WebSocket struct {
+	ServerIdentity *network.ServerIdentity
+	services       map[string]Service
+	server         *graceful.Server
+	mux            *http.ServeMux
+}
+
+// NewWebSocket opens a webservice-listener one port above the given
+// ServerIdentity.
+func NewWebSocket(si *network.ServerIdentity) *WebSocket {
+	w := &WebSocket{
+		ServerIdentity: si,
+		services:       make(map[string]Service),
+	}
+	webHost, err := getWebHost(w.ServerIdentity)
+	log.ErrFatal(err)
+	w.mux = http.NewServeMux()
+	w.server = &graceful.Server{
+		Timeout: 100 * time.Millisecond,
+		Server: &http.Server{
+			Addr:    webHost,
+			Handler: w.mux,
+		},
+	}
+	log.Lvl1("Starting to listen on", webHost)
+	go w.server.ListenAndServe()
```

Comment:
I would put the separate function that starts the websocket server  in `conode.Start()`. 
Like you, I have things that I do like to see, and being able to add `handlers` while the server is already running is one of those things. Imagine a client try to contact a conode exactly just before the URL is registered etc etc. It just seems simpler to register everything beforehand, even print those paths, and then start it.

Although the documentation does not say whether it's safe to add handlers at "run" time, all examples one could find on internet simply register everything before calling `http.ListenAndServe`. I'd advise to use the best practices as always.